platform/addons.cue

```diff
@@ -1,43 +1,39 @@
 package holos
-import ("encoding/json")
 
+import ("encoding/json")
 
-Addons: {
-  "cilium": {
-    path: "addons/cni/cilium"
-    schema: {
-      fleet: {
-        name: string
-      }
-      addonParam: string
-    }
-    parameters: {
-      addonParam: "test"
-    }
-    selector: {
-      "cluster_cni": "cilium"
-    }
-  }
+// hidden because parameters is not concrete.
+_Addons: {
+	"cilium": {
+		path:       "addons/cni/cilium"
+		parameters: #CiliumParameters
```

Comment:
Platform validates against the same interface here.

Note, this is quite awkward...  I made `_Addons` a hidden field because parameters isn't concrete now:

```shell
holos cue eval -e _Addons ./platform
```
```cue
cilium: {
    path: "addons/cni/cilium"
    parameters: {
        fleet: {
            name:   string
            meshed: bool
        }
        cluster: {
            name: string
            os:   string
        }
    }
    selector: {
        cluster_cni: "cilium"
    }
}
```